server.js

```diff
@@ -1,38 +1,89 @@
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
-
-const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo"
-mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
-mongoose.Promise = Promise
-
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
-const port = process.env.PORT || 8080
-const app = express()
-
-// Add middlewares to enable cors and json body parsing
-app.use(cors())
-app.use(bodyParser.json())
-
-// Start defining your routes here
-app.get('/', (req, res) => {
-  res.send('Hello world')
-})
-
-// Start the server
+import express from "express";
+import bodyParser from "body-parser";
+import cors from "cors";
+import mongoose from "mongoose";
+
+import albumData from "./data/rolling-stone-top-500-albums.json";
+
+const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo";
+mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true });
+mongoose.Promise = Promise;
+
+const Album = new mongoose.model("Album", {
+  position: Number,
+  artist: String,
+  albumName: String,
+  label: String,
+  year: Number,
+  critic: String,
+});
+
+if (process.env.RESET_DATABASE) {
+  const seedDatabase = async () => {
+    await Album.deleteMany(); // deleteMany and save are methods provided by Mongoose package
+
+    albumData.forEach(item => {
+      const newAlbum = new Album(item);
+      newAlbum.save();
+    });
+  };
+  seedDatabase();
+}
+
+const port = process.env.PORT || 8082;
+const app = express();
+const listEndpoints = require("express-list-endpoints");
+
+app.use(cors());
+app.use(bodyParser.json());
+
+app.use((req, res, next) => {
+  if (mongoose.connection.readyState === 1) {
+    next();
+  } else {
+    res.status(503).json({ error: `Service unavailable` });
+  }
+});
+
+app.get("/", (req, res) => {
+  res.send(listEndpoints(app));
+});
+
+app.get("/albums", async (req, res) => {
+  const allAlbums = await Album.find(req.query); // Clean way of setting queries. This allows all database fields to be queries
+  if (allAlbums) {
+    res.json({
+      numberOfResults: allAlbums.length,
+      results: allAlbums,
+    });
+  } else {
+    res.status(404).send({ error: `No albums found` });
+  }
+});
+
+app.get("/albums/:position", async (req, res) => {
+  const { position } = req.params;
+  const singleAlbum = await Album.findOne({ position: +position });
+  if (singleAlbum) {
+    res.json(singleAlbum);
+  } else {
+    res.status(404).send({ error: `No album found for position: ${position}` });
+  }
+});
+
+app.get("/albumid/:id", async (req, res) => {
+  try {
```

Comment:
Here you are using try/catch, and in the endpoint above you are using if/else. Is there a reason? If not -maybe you could be consistent in only using one of the kinds?

Reply:
Agreed. I feel like I have a better understanding of this now